code/Controllers/CMSMain.php

```diff
@@ -2161,10 +2135,8 @@ public function BatchActionList()
 
     /**
      * @deprecated 5.0 Please use custom logic for this
-     * @param $request
-     * @return HTTPResponse|string|void
      */
-    public function publishall($request)
+    public function publishall(HTTPRequest $request): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array